Tests/Unit/Http/RuleMatcherTest.php

```diff
@@ -45,4 +45,16 @@ public function testMatchResponse()
         $this->assertFalse($ruleMatcher->matches(new Request(), new Response('', 100)));
         $this->assertTrue($ruleMatcher->matches(new Request(), new Response('', 500)));
     }
+
+    public function testMatchResponseRequest()
+    {
+        $ruleMatcher = new RuleMatcher(new RequestMatcher(), array('match_response' => 'response.getStatusCode() == 201 & request.getMethod() == \'PUT\''));
```

Comment:
could this not be solved with `new RuleMatcher(new RequestMatcher(null, null, 'PUT'), array('match_response' => 'response.getStatusCode() == 201))` ?

i am reluctant to add redundant possibilities to achieve the same, as that is always confusing.